test/testinfra.py

```diff
@@ -343,11 +367,18 @@ def scan(self, update, context, except_context=False):
         results = []
         master_contexts = []
         pull_contexts = []
+        verify_contexts = []
+        image_contexts = []
```

Comment:
I'm uncomortable merging this all into one monster function. There are several things that need to be worked out here:

 * Separating the code into more manageable chunks somehow.
 * Returning something other than a tuple, a GithubTask() object